wordpress-mobile/WordPress-Android
# [Lint Warnings] Reconfigure & Add Baseline

Parent #18142
Closes #18161 

This PR changes Lint's (outdated) configuration ([lint/lint.xml](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/7bbad9625f2c27739f150bbe486ac4714dafb061)), enables all Lint warnings as errors ([warningsAsErrors = true](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/dc09bba7da6d6ca57339eb7785c117fd6fc9d4de)) and adds a new Lint baseline ([lint/baseline.xml](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/dd6610f6ffa22c76d144225874b5668ff865682f)).

FYI: With this change this project is effectively becomes more strict on Lint by not allowing any more warnings entering the codebase. Also, with that, a baseline of about `500` existing Lint warnings is created, which we are going to deal with (resolve/suppress) at a later point in time, most probably via #18162.

-----

Reconfigure Lint (Remove):

1. [Remove unnecessary icon missing density folder info config.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/324aec14d9268b7a9ec54b9dd640157f6f445224)
2. [Remove unnecessary rtl symmetry warning config.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/37e9642f8ec62ca481c4f06e4b2d887fff088c00)
3. [Remove unnecessary use sparse arrays warning config.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/5631527013a6b97138bac31e1c4292bfcddd91b4)
4. [Remove unnecessary label for ignore config.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/ab8d180305fdbbe367a0f00853a79ea4c92d0626)
5. [Remove unnecessary old target api ignore config.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/7e74e9d130f646ed8e69ee58cde9459d1ac99fc4)
6. [Remove use app tint ignore config and use default.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/49284ed8f2b8de5ceaaa79a6f6ca27f4df2b419d)
7. [Remove unnecessary invalid package config.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/8bc2bccf9fdb69f0b7a8ec9fc69e5653713224de)
8. [Remove unnecessary unused res ignore config for gradle caches.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/ef1c9d7c769fa7dabb4ba51a59ca32149845c3b3)
9. [Remove unnecessary invalid fragment version ignore config.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/7d7274b13730b1c15067a9bfea58be4e9e2ae443)

Reconfigure Lint (Narrow Down):

1. [Narrow down ignore config for icon dip size.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/8c9975cf86fd4bdd75fb6eeac24f63106dbf9249)
2. [Narrow down error config for unused resources.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/9246150ef1929575ae1e5a8a9c530488ea9fdd46)
3. [Narrow down error config for unused resources for generated.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/45ea4ad84f64c2b7c06d448b60358fc469c08146)
4. [Ignore unused resources lint warnings for generated.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/ce8fec94e77007b8c01d84d486f573ab31e2809e)

Reconfigure Lint (Reformat):

1. [Reorder lint config issues based on lint category.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/76cac9487ada602b728fba01a7b86bdc2fc38796)
2. [Add lint category on lint config issues.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/599c268f388cb34e57c506021242450cabe3ba5b)
3. [Add another layer of lint category on lint config issues.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/7bbad9625f2c27739f150bbe486ac4714dafb061)

Reconfigure Lint (Ignore):

1. [Ignore missing firebase instance token refresh lint warning.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/8f2eba1118c561ac086cb094a3037a8be22117c5) -> #18185
2. [Ignore gradle dependency lint warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/93162b39942a2cfa51e42a7817b6491f8eece880)
3. [Ignore icon missing density folder lint warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/91d66d5156f10de477c4191cdca4db94c25d39c4)
4. [Ignore trust all x509 trust manager lint warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/f0ab27572a4424679076ba81cabbe0ddebd82d5c)

Warnings Suppression List:

1. [Suppress app bundle locale changes lint warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/a84b3a94acd7909d50f012c414bcc933d650d661) -> #18188

Warnings Resolution List:

1. [Resolve ignore without reason lint warning.](https://github.com/wordpress-mobile/WordPress-Android/pull/18186/commits/621687292f184546b07b3c56659b9e4b7aac36c9)

-----

PS: @AjeshRPai I added you as the main reviewer, randomly, since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid. I also added the @wordpress-mobile/apps-infrastructure team, but this in done only for monitoring purposes, as such, I am not expecting any active review from that team. Thus, feel free to merge this PR if you deem so.

-----

## To test:

1. There is nothing much to test here.
2. Verifying that all the CI checks are successful should be enough.
3. However, if you want to be thorough about reviewing this change, you could try and run Lint locally on both, the `WordPress` and `Jetpack` apps, on all, the `Jalapeno`, `Wasabi` and `Vanilla` product flavors, on both, the `Debug` and `Release` build types, and see if everything is working as expected.

-----

## Merging Instructions:

1. [x] Wait for this PR to be reviewed, tested and approved.
2. [x] Merge this PR with latest `trunk` and wait for CI to run again all its checks.
3. [x] If CI checks are all ✅ then remove the `[Status] Not Ready for Merge` label and merge to `trunk`.
4. [x] Else, suppress or resolve any new Lint warnings that might appear and try this whole process again.

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

5. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

6. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:
